.github/workflows/messaging.yml

```diff
@@ -0,0 +1,67 @@
+name: messaging
+
+on:
+  pull_request:
+    paths:
+    - 'Firebase/Messaging/**'
+
+    #Podspec
+    - 'FirebaseMessaging.podspec'
+
+    - '.github/workflows/messaging.yml'
+  schedule:
+    # Run every day at 11pm (PST) - cron uses UTC times
+    - cron:  '0 7 * * *'
+
+jobs:
+  messaging:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Setup Bundler
+      run: scripts/setup_bundler.sh
+
+    - name: BuildAndTest # can be replaced with pod lib lint with CocoaPods 1.10
+      run: scripts/third_party/travis/retry.sh scripts/build.sh Messaging all
+
+  pod_lib_lint:
+    runs-on: macOS-latest
+
+    strategy:
+      matrix:
+        target: [ios, tvos, macos]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Bundler
+      run: scripts/setup_bundler.sh
+    - name: Build and test
+      run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseMessaging.podspec --platforms=${{ matrix.target }}
```

Comment:
it might be possible to add watchos to the matrix here with something like this:
```
if: matrix.target == 'watchos'
      run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseMessaging.podspec --skip-tests --platforms=${{ matrix.target }}
if: matrix.target == 'watchos'
      run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseMessaging.podspec --platforms=${{ matrix.target }}
```

Reply:
I wonder if the second condition will not happen if the first one is not passed. 
looking at the if: github.event_name == 'schedule'

Reply:
Totally possible! I suspect it can be done, but my knowledge of this syntax is so low, I'm not a good resource. Feel free to change it back if it's too much trouble.